README.md

```diff
@@ -30,25 +54,225 @@ npm install --save ipfsd-ctl
+local.spawn(function (err, ipfsd) {
+  const ipfsCtl = ipfsd.ctl
+  const ipfsCtrl = ipfsd.ctrl
+  ipfsCtl.id(function (err, id) {
     console.log(id)
-    process.exit()
+    ipfsCtrl.stopDaemon()
   })
 })
 ```
 
-If you need want to use an existing ipfs installation you can set `$IPFS_EXEC=/path/to/ipfs` to ensure it uses that.
+### Remote node
+
+```js
+// Start a remote disposable node, and get access to the api
+// print the node id, and stop the temporary daemon
+
+// IPFS_PATH will point to /tmp/ipfs_***** and will be
+// cleaned up when the process exits.
+
+const daemonFactory = require('ipfsd-ctl')
+const server = daemonFactory.server
+
+server.start((err) => {
+  if (err) {
+    throw err
+  }
+  
+  const remote = daemonFactory.remoteController(port || 9999)
+  remote.spawn(function (err, controller) {
+    const ipfsCtl = controller.ctl
+    const ipfsCtrl = controller.ctrl
+    ipfsCtl.id(function (err, id) {
+      console.log(id)
+      ipfsCtrl.stopDaemon()
+      server.stop()
+    })
+  })  
+})
+```
+
+It's also possible to start the server from `.aegir` `pre` and `post` hooks. 
+
+```js
+'use strict'
+
+const server = require('./src').server
+
+module.exports = {
+  karma: {
+    files: [{
+      pattern: 'test/fixtures/**/*',
+      watched: false,
+      served: true,
+      included: false
+    }],
+    singleRun: true
+  },
+  hooks: {
+    browser: {
+      pre: server.start,
+      post: server.stop
+    }
+  }
+}
+```
+
+## API
+
+### Daemon Factory
+
+#### Create factory
+
+- `daemonFactory.localController` - create a local controller
+- `daemonFactory.remoteController([port])` - create a remote controller, usable from browsers
+  - These methods return a factory that exposes the `spawn` method, which allows spawning and controlling ipfs nodes
+- `daemonFactory.server` - exposes `start` and `stop` methods to start and stop the bundled http server that is required to run the remote controller.
+
+#### Spawn nodes
+
+> Spawn either a js-ipfs or go-ipfs node through `localController` or `remoteController`
+
+`spawn([options], cb)`
+
+- `options` - is an optional object with various options and ipfs config parameters
+  - `js` bool (default false) - spawn a js or go node (default go)
+  - `init` bool (default true) - should the node be initialized
+  - `start` bool (default true) - should the node be started
+  - `repoPath` string - the repository path to use for this node, ignored if node is disposable
+  - `disposable` bool - a new repo is created and initialized for each invocation
+  - `args` - array of cmd line arguments to be passed to ipfs daemon
+  - `config` - ipfs configuration options
+  
+ - `cb(err, {ctl: <ipfs-api instance>, ctrl: <Node (ctrl) instance>})` - a callback that receives an object with two members:
+   - `ctl` - an [ipfs-api](https://github.com/ipfs/js-ipfs-api) instance attached to the newly created ipfs node
+   - `ctrl` - an instance of a daemon controller object
```

Comment:
```
- callback is a function with signature function (err, ipfsd) where:
  - err is an error in case any problem happens with the node spawning
  - ipfsd is an object that has two properties:
    - ipfsd.ctl - ...
    - ipfsd.ctrl - ...
```